### DISCUSSION TITLE
When closing last tab, have an option to replace it with a new tab without closing the browser

### DISCUSSION THREAD

**Discussion by sirinath** [zen-browser/desktop#1086] [Ideas]
When closing last tab, have an option to replace it with a new tab without closing the browser

**Comment by Neintonine** [comment]
Or at least having the option to enable that behaviour. Its quite annoying if you listen to music in the sidebar and then the browser closes, when you want to cleanup your tabs.

Isn't there something in Firefox for that or am I misremembering?

**Reply by sirinath** [reply]
I am not sure about Firefox but it is there Vivaldi Browser.

**Comment by rReyvn** [comment]
set browser.tabs.closeWindowWithLastTab to false in about:config

**Reply by alissonjferreira** [reply]
This worked, but there is a small bug: when closing the last tab, two new ones are created, when there should be only one

**Reply by JSGRD22** [reply]
> This worked, but there is a small bug: when closing the last tab, two new ones are created, when there should be only one

I remember it used to work properly, but after a new update this started happening

**Reply by Lampe2020** [reply]
This does not work for me somehow. I looked up the setting, it's set to true, but Zen still opens a new tab when I close the last one instead of closing the window. Setting it to false didn't change that and setting it back to true didn't remind the browser that it should close windows with no tabs in them…   
[EDIT]: Interestingly enough, in private windows it works. Closing the last tab closes the window, but not in a normal window.

**Reply by PhilipSchaefer86** [reply]
> This does not work for me somehow. I looked up the setting, it's set to true, but Zen still opens a new tab when I close the last one instead of closing the window. Setting it to false didn't change that and setting it back to true didn't remind the browser that it should close windows with no tabs in them…

same for me :(

**Reply by jakelauer** [reply]
Any way to fix this? Quite strange to have a window stick around after Cmd+W.

**Reply by PhilipSchaefer86** [reply]
For me it is fixed in the Zen Twilight release.

**Reply by eden881** [reply]
Thanks! Worked for me on `1.7b`

**Reply by bachjessen** [reply]
@eden881 do you have essential tabs, i have mine set to true but can't close the windows when i try to close the "new tab" tab and continue cmd+w

**Reply by eden881** [reply]
@simon-bd hmm.. It looks like you're right. Even when I enable this setting again my browser doesn't close now.
It did before though...
I do have essential tabs but I also had them when the browser closed for me.

**Reply by Lampe2020** [reply]
> This does not work for me somehow. […]

Now in v1.7.6b it works. `browser.tabs.closeWindowWithLastTab` was set to `false` again because I had deleted my Zen profile folder by accident and had to restore its contents using Firefox's sync functionality. Setting it to `true` finally made Zen close the window when the last tab is closed.

**Comment by 0x6e7a6f** [comment]
Yes like Vivaldi or Arc (and other). I love this feature, this is really annoying to be obligated to open a blank page before closing the last tab.

**Comment by AlJohri** [comment]
I set `browser.tabs.closeWindowWithLastTab` to true and I still can't get it to close the window when the last tab is closed.

Related: https://www.reddit.com/r/zen_browser/comments/1h646ec/zen_not_quitting_when_closing_last_tab/
